build: Bump minimum supported Clang to clang-10

maflcko opened the pull request:
It doesn't make sense to support a minimum clang version that is difficult to install on all supported operating systems, which generally ship a later version:

* Ubuntu Focal 20.04: https://packages.ubuntu.com/focal/clang-10 and https://packages.ubuntu.com/focal/clang-12
* Debian Bullseye: https://packages.debian.org/bullseye/clang-13
* CentOS 8 Stream: All Clang versions from 11.0 to 15.0

Also, it allows to drop build code, which means it won't waste review when rolling over into cmake (`cmake/module/CheckStdFilesystem.cmake`).

Commit:
doc: Remove outdated CentOS comment

No longer applicable after CONFIG_SHELL must be explicitly set to dash
after commit fafc55a489d8f654dd7732b699c9bf51b5136b4b.

Commit:
build: Bump minimum Clang to clang-10

hebasto commented:
> Also, it allows to drop build code, which means it won't waste review when rolling over into cmake (`cmake/module/CheckStdFilesystem.cmake`).

Concept ACK.

hebasto commented on build-aux/m4/l_filesystem.m4:
```diff
@@ -31,15 +30,8 @@ AC_DEFUN([CHECK_FILESYSTEM], [
-        ],[
-          AC_MSG_RESULT([no])
-          AC_MSG_CHECKING([whether std::filesystem needs -lc++fs])
-          LIBS="$SAVED_LIBS -lc++fs"
```
The only left usage of `SAVED_LIBS` can de dropped as well:
```diff
--- a/build-aux/m4/l_filesystem.m4
+++ b/build-aux/m4/l_filesystem.m4
@@ -25,8 +25,7 @@ AC_DEFUN([CHECK_FILESYSTEM], [
       AC_MSG_RESULT([yes])
     ],[
       AC_MSG_RESULT([no])
-      SAVED_LIBS="$LIBS"
-      LIBS="$SAVED_LIBS -lstdc++fs"
+      LIBS="$LIBS -lstdc++fs"
       AC_MSG_CHECKING([whether std::filesystem needs -lstdc++fs])
       AC_LINK_IFELSE([AC_LANG_SOURCE([_CHECK_FILESYSTEM_testbody])],[
           AC_MSG_RESULT([yes])
```

hebasto commented:
Just noting that our current Guix environment uses Clang 10.0.1.

Commit:
ci: Drop NO_WERROR=1 for clang-10 build

This partially reverts commit 71383f2fad065378393ef55b6d65e14c656b7301.
This should be fine, because if warnings are issues again in the future,
it can be disabled again, along with a list of the false warnings.

hebasto reviewed (approved):
ACK fa199ee614a7ed99c6caf329093a3573ea5a664b

fanquake reviewed (approved):
ACK fa199ee614a7ed99c6caf329093a3573ea5a664b
